test/broadway_test.exs

```diff
@@ -487,6 +481,69 @@ defmodule BroadwayTest do
       assert named_priority(get_processor(broadway_name, :default)) == {:priority, :high}
     end
 
+    test "supports starting under a via tuple" do
+      defmodule MyRegistry do
+        def start_link do
+          Registry.start_link(keys: :unique, name: __MODULE__)
+        end
+
+        def via_tuple(name) do
+          {:via, Registry, {__MODULE__, name}}
+        end
+
+        def whereis_name(name) do
+          Registry.whereis_name({__MODULE__, name})
+        end
+      end
```

Comment:
I believe we don't need to create this module, instead, just use `Registry` directly?